src/definition/populate_test.go

```diff
@@ -456,75 +421,97 @@ func TestIntegrationProtocol2PopulateFunc_PopulateOnlySpecifiedGroups(t *testing
-	expectedEntityData3, err := sdk.NewEntityData("playground", "k8s:cluster")
-	if err != nil {
-		t.Fatal()
+	expectedMetricSet2 := &metric.Set{
+		Metrics: map[string]interface{}{
+			"event_type":  "TestSample",
+			"metric_1":    float64(3),
+			"metric_2":    float64(4),
+			"displayName": "testEntity12-generated",
+			"clusterName": "playground",
+		},
 	}
-	expectedMetricSet3 := metric.MetricSet{
-		"event_type":        "K8sClusterSample",
-		"entityName":        "k8s:cluster:playground",
-		"clusterName":       "playground",
-		"clusterK8sVersion": "v1.15.42",
+	expectedEntityData2.Metrics = []*metric.Set{expectedMetricSet2}
+
+	expectedEntityData3, err := dummyIntgr.Entity("playground", "k8s:cluster")
+	require.NoError(t, err)
+
+	expectedMetricSet3 := &metric.Set{
+		Metrics: map[string]interface{}{
+			"event_type":        "K8sClusterSample",
+			"clusterName":       "playground",
+			"clusterK8sVersion": "v1.15.42",
+		},
 	}
-	expectedEntityData3.Metrics = []metric.MetricSet{expectedMetricSet3}
-	expectedInventory := sdk.Inventory{}
-	expectedInventory.SetItem("cluster", "name", "playground")
-	expectedInventory.SetItem("cluster", "k8sVersion", "v1.15.42")
+	expectedEntityData3.Metrics = []*metric.Set{expectedMetricSet3}
+	expectedInventory := inventory.New()
+	_ = expectedInventory.SetItem("cluster", "name", "playground")
+	_ = expectedInventory.SetItem("cluster", "k8sVersion", "v1.15.42")
 	expectedEntityData3.Inventory = expectedInventory
 
-	integration, err := sdk.NewIntegrationProtocol2("nr.test", "1.0.0", new(struct{}))
-	if err != nil {
-		t.Fatal()
-	}
-	populated, errs := IntegrationProtocol2PopulateFunc(
-		integration,
+	dummyIntgr.Clear()
+	populated, errs := IntegrationPopulator(
+		dummyIntgr,
 		defaultNS,
 		&version.Info{GitVersion: "v1.15.42"},
 		fromGroupMetricSetTypeGuessFunc,
-		metricsNamingManipulator,
-		clusterMetricsManipulator,
 	)(groups, withGeneratorSpec)
 	assert.True(t, populated)
 	assert.Empty(t, errs)
-	assert.Contains(t, integration.Data, &expectedEntityData1)
-	assert.Contains(t, integration.Data, &expectedEntityData2)
-	assert.Contains(t, integration.Data, &expectedEntityData3)
-	assert.Len(t, integration.Data, 3)
+	assert.Len(t, dummyIntgr.Entities, 3)
+
+	// Search for missing entities using cmp
+	compareIgnoreFields := cmpopts.IgnoreUnexported(integration.Entity{}, metric.Set{}, inventory.Inventory{})
+	for _, ti := range []*integration.Entity{
+		expectedEntityData1,
+		expectedEntityData2,
+		expectedEntityData3,
+	} {
+		found := false
+		closestMatchDiff := ""
+		for _, presentEntity := range dummyIntgr.Entities {
+			curDiff := cmp.Diff(presentEntity, ti, compareIgnoreFields)
+			// If curDiff is empty we got an exact match, return.
+			if len(curDiff) == 0 {
+				found = true
+				break
+			}
+
+			// Otherwise, store current diff as closest match if it is smaller than the closestMatch, or if the latter is empty
+			if len(closestMatchDiff) == 0 || len(curDiff) < len(closestMatchDiff) {
```

Comment:
Ditto `""` over `len() == 0`. Let's leave `len()` for slices or maps when possible :)

Reply:
Nice point